### PR TITLE
Add an event to distinguish failures that are not final

### PR DESCRIPTION
**Pull request by bhtucker**
A proposed implementation to resolve https://github.com/harrystech/arthur-redshift-etl/issues/72

User facing change:
* Monitor payloads' "event" field may now have a new value, `'fail_before_retry'`, that indicates a step attempt ended in error but will be retried

Internal changes:
* Concurrent load event polling now looks for only the events it can handle ('fail' and 'finish' events) rather than 'everything but start'

(HA-733)

### CODE REVIEW

File: python/etl/extract/extractor.py [hunk 1]
```diff
@@ -77,6 +77,7 @@ def extract_source(self, source: DataWarehouseSchema,
         with Timer() as timer:
             for i, relation in enumerate(relations):
                 try:
+                    retries = get_config_int("arthur_settings.extract_retries")
```
Comment (tvogels01) [inline python/etl/extract/extractor.py]: This is duplicate code (the old line still exists below the def) ... Please move this lookup of a runtime config outside the loop (and have it only once ;-)

File: python/etl/monitor.py [hunk 2]
```diff
@@ -47,6 +47,7 @@
 STEP_START = "start"
 STEP_FINISH = "finish"
 STEP_FAIL = "fail"
+STEP_ATTEMPT_FAIL = "attempt_fail"
```
Comment (tvogels01) [inline python/etl/monitor.py]: Maybe "fail_before_retry"?

**Comment by tvogels01** [comment]
I think we get ourselves into trouble in load.py. When we receive a "fail before retry" event, then we'd try to load it here: https://github.com/harrystech/arthur-redshift-etl/blob/next/python/etl/load.py#L720

We can either keep the current semantics and treat fail and "fail before retry" the same way. Or simply ignore for now the "fail before retry" events in load.

**Comment by bhtucker** [comment]
Indeed — added a commit to filter explicitly for the events that we handle.

**Pull request merged**